pkg/api/convertertoapi.go

```diff
@@ -140,7 +140,13 @@ func convertV20160930Properties(v20160930 *v20160930.Properties, api *Properties
 	api.AgentPoolProfiles = []AgentPoolProfile{}
 	for _, p := range v20160930.AgentPoolProfiles {
 		apiProfile := &AgentPoolProfile{}
-		convertV20160930AgentPoolProfile(&p, apiProfile)
+		// api.OrchestratorProfile already be filled in correctly
+		if api.OrchestratorProfile.IsKubernetes() {
+			convertV20160930AgentPoolProfile(&p, AvailabilitySet, apiProfile)
+
+		} else {
```

Comment:
this should have a comment. about for this api version we only allowed these avaiblability types for these orchestrators